### ISSUE TITLE
Plugin v2019.12.02 does not load into IntelliJ 2019.3.1

### ISSUE THREAD

**Issue by konste** [bazelbuild/intellij#1522]
When I try to download the plugin version v2019.12.02 (latest available at the moment in the Gallery) I get `Plugin 'Bazel' is incompatible with this installation. `
IntelliJ Ultimate 2019.3.1
Plugin v2019.12.02
Bazel 2.0.0
How do I figure out what exactly is not compatible and what shall I upgrade/downgrade to become compatible?

**Comment by Saser** [comment]
If you look at the [versions list](https://plugins.jetbrains.com/plugin/8609-bazel/versions) for the plugin, you can see that plugin version v2019.12.02.0.0 is compatible with IntelliJ IDEA Ultimate 2019.2.4, and not 2019.3.1. This is most likely the reason for your error message.

I had a similar situation, and managed to resolve it by building the plugin from source, following these steps:
1. Clone this repository.
1. Check out the tag `v2019.12.02`.
1. Build the plugin using the instructions supplied in the README. Specifically, I issued the following command:
    ```shell
    bazel build //ijwb:ijwb_bazel_zip --define=ij_product=intellij-ue-2019.3
    ```
    I found the value for `ij_product` by looking at the file [`intellij_platform_sdk/build_defs.bzl`](https://github.com/bazelbuild/intellij/blob/v2019.12.02/intellij_platform_sdk/build_defs.bzl#L46-L49). The README [mentions this file](https://github.com/bazelbuild/intellij/blob/v2019.12.02/README.md#building-the-plugin).
1. Install the plugin in IntelliJ. Go to `File > Settings > Plugins`. From the little wheel in the top of this window, choose `Install Plugin from Disk...`. Locate the .zip file built above -- it should be in `bazel-bin/ijwb`.

Hopefully this can help you too!

---

I should note that I had some trouble building the plugin initially. It turned out to be a problem with `@local_jdk` -- my system uses OpenJDK 11, which seems to be missing the file `$JAVA_HOME/lib/tools.jar`. I resolved it by installing OpenJDK 8 from my operating system's package manager, and modifying the `bazel build` command above:
```shell
JAVA_HOME=/path/to/jdk8 bazel build //ijwb:ijwb_bazel_zip --define=ij_product=intellij-ue-2019.3
```

**Comment by konste** [comment]
Thanks a lot @Saser! This is really helpful. I will give it a try on Monday.

**Comment by konste** [comment]
@Saser So I can declare partial success. I was able to build, load and use newer version of the plugin with IntelliJ Idea Ultimate. Lessons learned: 1. It is important to checkout to the tag - HEAD is not good. 2. Using OpenJDK 8 is important, JDK 11 would not work.

I also tried to build the version for CLion with `bazel build //clwb:clwb_bazel_zip --define=ij_product=clion-2019.3` but it causes Internal error in Bazel during the build! I've tried to load IntelliJ plugin to CLion and it loads, but unfortunately does not work.

Another bummer is that debugger does not stop on the breakpoints, which actually was the purpose of the whole exercise. I am on Windows and it seems to somehow make the debugger to run the build without errors, but... no stopping on breakpoints. Crying...
